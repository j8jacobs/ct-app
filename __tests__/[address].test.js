import React from "react";
import renderer from "react-test-renderer";
import AddressPage from "../app/address/[address]";
import { useLocalSearchParams } from "expo-router";

describe("<AddressPage />", () => {
  beforeEach(() => {
    useLocalSearchParams.mockReturnValue({
      address: "3E8ociqZa9mZUSwGdSmAEMAoAxBK3FNDcd",
      final_balance: 1.0001,
      n_tx: 50,
    });
  });
  it("renders", () => {
    const tree = renderer.create(<AddressPage />).toJSON();
    expect(tree.children.length).toBe(3);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<AddressPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
