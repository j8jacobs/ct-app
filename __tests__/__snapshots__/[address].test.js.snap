// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddressPage /> renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  MockScreen
  <View
    style={
      {
        "alignItems": "center",
        "height": 100,
        "justifyContent": "center",
        "position": "relative",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 32,
        }
      }
    >
      $
      0
    </Text>
    <View
      style={
        {
          "height": "100%",
          "justifyContent": "center",
          "position": "absolute",
          "right": 12,
        }
      }
    >
      <Text
        allowFontScaling={false}
        onPress={[Function]}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        î¸¢
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#f0f0f0",
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 8,
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        getItem={[Function]}
        getItemCount={[Function]}
        initialNumToRender={5}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onRefresh={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        refreshControl={
          <RefreshControlMock
            onRefresh={[Function]}
            refreshing={true}
          />
        }
        refreshing={true}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={0.0001}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <RCTRefreshControl />
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
